creed/tiling_utils_test.py

```diff
@@ -280,6 +315,75 @@ def test_generate_x_y_fov_pairs():
     assert sample_pairs == [(0, 2), (0, 4), (5, 2), (5, 4)]
 
 
+# TODO: clean up this test, something like Adam's setup might work here
+@pytest.mark.parametrize('rhombus_test_name', list(_TMA_RHOMBUS_TEST_NAMES.keys()))
+def test_generate_x_y_fov_pairs_rhombus(rhombus_test_name):
```

Comment:
the pytest-cases setup would let you move a lot of the `if rhombus_test_name == ...` logic into each individual test case.  That looks like:

```
from pytest_cases import parametrize_with_cases

X_COORDS = (1500, 1666, 1833, 2000)

class RhombusTestCoords:

    def case_rectangle():
        coords = [
                XYCoord(1500, 2000),
                ...
        ]
        actual_pairs = []
        for x_coord in XCOORDS:
                actual_pairs.extend(...)

        return coords, actual_pairs

    def case_xslant():
         ...
...

@parametrize_with_cases('coords, pairs', cases=RhombusTestCoords)
def test_generage_x_y_fov_pairs_rhombus(coords, pairs):
    top_left, top_right, bottom_left, bottom_right = coords

    # generate the FOV-coordinate pairs
    pairs = tiling_utils.generate_x_y_fov_pairs_rhombus(
        top_left, top_right, bottom_left, bottom_right, 4, 3
    )

    assert pairs == actual_pairs
```